### PR TITLE
docs: add docusaurus docs with typedoc

### PR DESCRIPTION
**Pull request by simoneb**
Spike to add documentation and API docs website via docusaurus and typedoc.

The website is built with a new GH workflow and automatically deployed to GitHub Pages, accessible at https://nearform.github.io/fast-jwt/ (the API section especially, which is auto generated by typedoc).

The documentation website simply contains the README of the repo and API documentation automatically generated via typedoc.

### TODO

- [ ] restrict the docs deployment to pushes to the master branch

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: docs
+
+on: [push]
```
Comment (Eomm) [inline .github/workflows/docs.yml:3]:
```suggestion
on:
  push:
    branches:
      - master
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,28 +1,24 @@
...
+      - uses: actions/cache@v2
         with:
           path: node_modules
           key: node-modules-${{ hashFiles('package.json') }}
-      - name: Install dependencies
-        run: npm install
-      - name: Run Tests
-        run: npm run ci
+      - run: npm install
+      - run: npm run ci
```
Comment (Eomm) [inline .github/workflows/ci.yml:21]: Could we rename this `script`?
I was confused at the first reading

```suggestion
      - run: npm run test:ci
```

File: .github/workflows/docs.yml (hunk 1 above)
Comment (simoneb) [reply .github/workflows/docs.yml:3]: see the todo in the PR description please

File: .github/workflows/ci.yml (hunk 2 above)
Comment (simoneb) [reply .github/workflows/ci.yml:21]: not in scope for this PR

**Pull request closed without merging**